plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunctionProcessor.java

```diff
@@ -0,0 +1,220 @@
+    private final FileFormatDataSourceStats fileFormatDataSourceStats;
+    private final ParquetReaderOptions parquetReaderOptions;
+    private final List<DeltaLakeColumnHandle> dataFilesColumns;
+    private final List<DeltaLakeColumnHandle> cdfFilesColumns;
+
+    private TableChangesFileType fileType;
+    private DeltaLakePageSource deltaLakePageSource;
+    private List<DeltaLakeColumnHandle> splitColumns;
+    private Block currentVersionAsBlock;
+    private long currentVersionCommitTimestamp;
+
+    public TableChangesFunctionProcessor(
+            ConnectorSession session,
+            TrinoFileSystemFactory fileSystemFactory,
+            DeltaLakeConfig deltaLakeConfig,
+            FileFormatDataSourceStats fileFormatDataSourceStats,
+            ParquetReaderOptions parquetReaderOptions,
+            TableChangesTableFunctionHandle handle)
+    {
+        this.session = requireNonNull(session, "session is null");
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.parquetDateTimeZone = deltaLakeConfig.getParquetDateTimeZone();
+        this.domainCompactionThreshold = deltaLakeConfig.getDomainCompactionThreshold();
+        this.fileFormatDataSourceStats = requireNonNull(fileFormatDataSourceStats, "fileFormatDataSourceStats is null");
+        this.dataFilesColumns = requireNonNull(handle, "handle is null").columns();
+        this.cdfFilesColumns = ImmutableList.<DeltaLakeColumnHandle>builder().addAll(handle.columns())
+                .add(new DeltaLakeColumnHandle(
+                        CHANGE_TYPE_COLUMN_NAME,
+                        VARCHAR,
+                        OptionalInt.empty(),
+                        CHANGE_TYPE_COLUMN_NAME,
+                        VARCHAR,
+                        REGULAR))
+                .build();
+        this.parquetReaderOptions = requireNonNull(parquetReaderOptions, "parquetReaderOptions is null")
+                .withMaxReadBlockSize(getParquetMaxReadBlockSize(session))
+                .withMaxReadBlockRowCount(getParquetMaxReadBlockRowCount(session))
+                .withUseColumnIndex(isParquetUseColumnIndex(session))
+                .withBatchColumnReaders(isParquetOptimizedReaderEnabled(session))
+                .withBatchNestedColumnReaders(isParquetOptimizedNestedReaderEnabled(session));
+    }
+
+    @Override
+    public TableFunctionProcessorState process(ConnectorSplit connectorSplit)
+    {
+        if (connectorSplit != null) {
+            TableChangesSplit split = (TableChangesSplit) connectorSplit;
+            fileType = split.fileType();
+            if (fileType == CDF_FILE) {
+                splitColumns = cdfFilesColumns;
+            }
+            else {
+                splitColumns = dataFilesColumns;
+            }
+            deltaLakePageSource = createDeltaLakePageSource(split);
+            currentVersionAsBlock = nativeValueToBlock(INTEGER, split.currentVersion());
```

Comment:
If it should be a long column:
```suggestion
            currentVersionAsBlock = nativeValueToBlock(BIGINT, split.currentVersion());
```